src/app-layer-ssl.c

```diff
@@ -1373,6 +1522,13 @@ static int TLSDecodeHandshakeHello(SSLState *ssl_state,
     int ret;
     uint32_t parsed = 0;
 
+    if (SC_ATOMIC_GET(ssl_config.enable_ja4) &&
+            ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO) {
+        ssl_state->curr_connp->ja4 = SCJA4New();
```

Comment:
Why is it this condition to prevent the leak ? `ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO`

Why not `ssl_state->curr_connp->ja4 == NULL`

Reply:
The idea is to only create a JA4 state if we have

1. JA4 effectively enabled, and
2. we are in a client hello (JA4 -- not JA4S -- is defined for the client only).

So just not having a JA4 state set yet is not enough to do the right thing. My assumption is there is only ever one handshake to be parsed in a TLS flow, and if there is a client hello, it is first.

You are right, I am not sure if `TLSDecodeHandshakeHello()` can be called multiple times in a flow for more than one client hello, so let's be defensive -- I have added an additional `NULL` check to avoid leaking memory by overwriting the pointer if it is already set, unlikely as it may be. 👍🏻